spec.html

```diff
@@ -4393,6 +4393,21 @@ <h1>The Lexical Environment and Environment Record Specification Types</h1>
       <p>The Lexical Environment and Environment Record types are used to explain the behaviour of name resolution in nested functions and blocks. These types and the operations upon them are defined in <emu-xref href="#sec-lexical-environments"></emu-xref>.</p>
     </emu-clause>
 
+    <emu-clause id="sec-abstract-closure">
+      <h1>The Abstract Closure Specification Type</h1>
+      <p>The <dfn>abstract closure</dfn> specification type is used to refer to algorithm steps together with a collection of values. Abstract closures are meta-values and are invoked using function application style such as _closure_(_arg1_, _arg2_). Like abstract operations, invocations perform the algorithm steps described by the abstract closure.</p>
+      <p>In algorithm steps that create an abstract closure, values are captured with the verb "capture" followed by a list of aliases. The value associated with each alias at creation time is captured. In steps that specify the algorithm to be performed when an abstract closure is called, each captured value is referred to by the alias that was used to capture the value.</p>
+      <p>If an abstract closure returns a Completion Record, that Completion Record's [[Type]] must be either ~normal~ or ~throw~.</p>
```

Comment:
```suggestion
      <p>If an abstract closure returns a Completion Record, that Completion Record's [[Type]] is either ~normal~ or ~throw~.</p>
```
(because "must be" suggests a requirement on an implementation, whereas this is just a self-imposed restriction on the spec)

Reply:
Disagree. There are no requirements on implementation anywhere when talking about specification types.

Reply:
Huh? I'm not even sure how that relates to what I said. Are you disagreeing with my statement that "must be" suggests a requirement on an implementation, or with my statement that the sentence in question is a self-imposed restriction on the spec?

> There are no requirements on implementation anywhere when talking about specification types.

If that were true, then when talking about specification types, we could say completely different things without changing any of the requirements on implementations. But that's clearly false. For example, imagine that, when talking about Lists, we said that elements are in an unpredictable order, or that duplicates are removed. That would certainly change the requirements that the spec imposes on implementations.

Reply:
The observable differences are requirements on implementations, anything else is purely editorial.

Reply:
Okay, but again, I'm not sure how that relates to my suggested change.